source/MaterialXCore/MaterialNode.cpp

```diff
@@ -0,0 +1,151 @@
+//
```

Comment:
Is this a good place to put material node logic?  There is no material element class to add methods to anymore and
this should belong in core and not shadergen as using just core means this functionality is no longer "available".

Reply:
Discussed with Jonathan Stone and this an suitable place for now.